migrations/800_groups__add.sql

```diff
@@ -0,0 +1,40 @@
+-- A Work in Progress to discuss shared schema for group storage in PL
+
+-- Abdu students called this group_type
+-- Suggestion: maybe call group_criteria or group_config or group_contraints?
+-- This group defines additional criteria or 'ownership' of the group, perhaps
+-- if the group was autogenerated based on rules
+--
+-- Haven't implemented this yet because it seems more about group creation or constraints
+-- on autogenerated groups. Because one group_config could be connected to multiple
+-- groups this is pulled out to a different table
+--CREATE TABLE IF NOT EXISTS group_config (
+--    id BIGSERIAL PRIMARY KEY,
+--    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
Weren't we planning to reference `assessments`?

Reply:
Did we end up deciding to make it course-wide and override those settings in each layer?

Reply:
There will probably need to be some linking between `assessments` and groups, probably in the `group_config` (or group_constraints, whatever we're going to call it). But I hadn't flushed that part out very much. A column might not be needed if the linkage comes from an assessment_access_rule?

Reply:
I agree we want a `group_configs.assessment_id` column.